### PR TITLE
Csghub  repo add to collections

### PR DESCRIPTION
**Pull request by ymh6315431**

### CODE REVIEW

File: app/javascript/components/collections/AddToCollections.vue [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        type: 'success'
+      })
+      collectionsIdsInput.value = ''
+    }
+  }
+  onMounted(() => {
+    fetchCollectionsList()
+  })
+</script>
+<style>
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:147]: should we add scoped?

File: app/javascript/components/collections/AddToCollections.vue [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    collectionsIdsInput.value = ''
+  }
+
+  async function repoAddToCollections() {
+    const addRepoData = {
+      repo_ids: [props.repoId]
+    }
+    const options = { method: 'POST', body: JSON.stringify(addRepoData) }
+    const url = `${csghubServer}/api/v1/collections/${collectionsIdsInput.value}/repos`
+    const response = await jwtFetch(url, options)
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:131]: please pull latest main branch code and update jwtFetch to the new method

File: app/javascript/components/collections/AddToCollections.vue (hunk 1 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: down

File: app/javascript/components/collections/AddToCollections.vue (hunk 2 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: down

File: app/javascript/components/collections/AddToCollections.vue [hunk 3]
```diff
@@ -0,0 +1,182 @@
+<template>
+  <div
+    class="AddRepoToCollections absolute top-0 right-[120px] md:relative md:right-0 md:pl-5 md:pb-4 z-20"
+  >
+    <div
+      v-if="settingsVisibility"
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:6]: update name from 'settingsVisibility' to 'canManage'

File: app/javascript/components/collections/AddToCollections.vue [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    collectionsIdsInput.value = ''
+  }
+
+  async function repoAddToCollections() {
+    const addRepoData = {
+      repo_ids: [props.repoId]
+    }
+    const options = { body: JSON.stringify(addRepoData) }
+    const url = `/collections/${collectionsIdsInput.value}/repos`
+    const { response, error } = await useFetchApi(url, options).post().json()
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:130]: const { data, error } = await useFetchApi(url, opions)

这我们直接取 data 保持统一？

Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue:130]: 这里解构赋值需要用 response

File: app/javascript/components/collections/AddToCollections.vue (hunk 3 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: done

File: app/javascript/components/collections/AddToCollections.vue [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    canManage: Boolean
+  })
+  const dialogVisible = ref(false)
+  const collectionsList = ref([])
+  const collectionsIdsInput = ref('')
+
+  const fetchCollectionsList = async () => {
+    const url = `/user/${props.userName}/collections`
+    const { data, error } = await useFetchApi(url).json()
+    console.log(data);
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:101]: remove

File: app/javascript/components/collections/AddToCollections.vue [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    const json = data.value
+    if (json) {
+      console.log(json);
```
Comment (hiveer) [inline app/javascript/components/collections/AddToCollections.vue:104]: remove

File: app/javascript/components/collections/AddToCollections.vue (hunk 4 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: done

File: app/javascript/components/collections/AddToCollections.vue (hunk 5 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: done

File: app/javascript/components/collections/AddToCollections.vue (hunk 6 above)
Comment (ymh6315431) [reply app/javascript/components/collections/AddToCollections.vue]: done

**Review by hiveer — APPROVED** [review]
LGTM

**Pull request merged**